File: code/get_all_file_name.m
% 获取全部文件夹内的文件
function file_list = get_all_file_name(path)
    dir_data = dir(path);
    file_list = dir_data(~[dir_data.isdir]);  % 当前文件夹下所有文件
    
    % 获取子文件夹
    sub_dir = dir_data([dir_data.isdir]);  % struct
    dot_dir = ismember({sub_dir.name}, {'.', '..'});  % logical
    sub_dir = sub_dir(~dot_dir);  % struct，不含特殊文件夹的所有子文件夹
    
    % 递归
    for i = 1:length(sub_dir)
        next_dir = fullfile(sub_dir(i).folder, sub_dir(i).name);  % str
        file_list = [file_list; get_all_file_name(next_dir)];  % struct
    end
end

File: code/python/get_all_file_name.py
from os import listdir, walk
from os.path import isfile, isdir, join


def get_all_file_name(path):

    file_list = [join(path, f) for f in listdir(path) if isfile(join(path, f))]
    sub_dir = [join(path, d) for d in listdir(path) if isdir(join(path, d))]

    for i in range(len(sub_dir)):
        next_dir = sub_dir[i]
        file_list.extend(get_all_file_name(next_dir))

    return file_list


def get_all_file_name_generator(path):

    for folder, subdirs, files in walk(path):
        for f in files:
            yield join(folder, f)
